sentry/src/main/java/io/sentry/Scopes.java

```diff
@@ -421,8 +422,9 @@ public void close(final boolean isRestarting) {
         }
 
         // TODO: should we end session before closing client?
-        // TODO [HSM] should we go through all clients (global, isolation, current) and close them?
-        getClient().close(isRestarting);
+        configureScope(ScopeType.CURRENT, scope -> scope.getClient().close(isRestarting));
```

Comment:
Closing clients on all scopes now.

Reply:
Do we ever have the case where multiple scopes (e.g. `global` and `isolation`) have the same client? If so, we would call close on the same `client` twice.

Reply:
This should only happen if a user sets the same client explicitly. Event then I think it should just work but we should test this scenario to know if it could cause problems.